backend/pkg/api/data_access/vdb_summary.go

```diff
@@ -114,113 +113,123 @@ func (d *DataAccessService) GetValidatorDashboardSummary(ctx context.Context, da
+		ds = ds.
+			SelectAppend(goqu.L("?::smallint AS result_group_id", t.DefaultGroupId)).
+			Where(goqu.L("r.validator_index IN ?", validators))
+	} else {
+		if dashboardId.AggregateGroups {
 			ds = ds.
-				SelectAppend(goqu.L("?::smallint AS result_group_id", t.DefaultGroupId)).
-				Where(goqu.L("r.validator_index = ANY(?)", pq.Array(validators)))
+				SelectAppend(goqu.L("?::smallint AS result_group_id", t.DefaultGroupId))
 		} else {
-			if dashboardId.AggregateGroups {
-				ds = ds.
-					SelectAppend(goqu.L("?::smallint AS result_group_id", t.DefaultGroupId))
-			} else {
-				ds = ds.
-					SelectAppend(goqu.L("v.group_id AS result_group_id"))
-			}
-
 			ds = ds.
-				InnerJoin(goqu.L("users_val_dashboards_validators v"), goqu.On(goqu.L("r.validator_index = v.validator_index"))).
-				Where(goqu.L("v.dashboard_id = ?", dashboardId.Id))
-
-			if groupNameSearchEnabled && (search != "" || colSort.Column == enums.VDBSummaryColumns.Group) {
-				// Get the group names since we can filter and/or sort for them
-				ds = ds.
-					SelectAppend(goqu.L("g.name AS group_name")).
-					InnerJoin(goqu.L("users_val_dashboards_groups g"), goqu.On(goqu.L("v.group_id = g.id AND v.dashboard_id = g.dashboard_id"))).
-					GroupByAppend(goqu.L("group_name"))
-			}
+				SelectAppend(goqu.L("v.group_id AS result_group_id"))
 		}
 
-		query, args, err := ds.Prepared(true).ToSQL()
-		if err != nil {
-			return fmt.Errorf("error preparing query: %v", err)
-		}
+		ds = ds.
+			InnerJoin(goqu.L("validators v"), goqu.On(goqu.L("r.validator_index = v.validator_index"))).
+			Where(goqu.L("r.validator_index IN (SELECT validator_index FROM validators)"))
 
-		err = d.alloyReader.SelectContext(ctx, &queryResult, query, args...)
-		if err != nil {
-			return fmt.Errorf("error retrieving data from table %s: %v", table, err)
+		if groupNameSearchEnabled && (search != "" || colSort.Column == enums.VDBSummaryColumns.Group) {
+			// Get the group names since we can filter and/or sort for them
+			ds = ds.
+				SelectAppend(goqu.L("g.name AS group_name")).
+				InnerJoin(goqu.L("users_val_dashboards_groups g"), goqu.On(goqu.L("v.group_id = g.id AND v.dashboard_id = g.dashboard_id"))).
+				GroupByAppend(goqu.L("group_name"))
 		}
-		return nil
-	})
+	}
 
+	query, args, err := ds.Prepared(true).ToSQL()
+	if err != nil {
+		return nil, nil, fmt.Errorf("error preparing query: %v", err)
+	}
+
+	err = d.clickhouseReader.SelectContext(ctx, &queryResult, query, args...)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error retrieving data from table %s: %v", clickhouseTable, err)
+	}
+
+	epochMin := int64(math.MaxInt64)
+	epochMax := int64(0)
+
+	for _, row := range queryResult {
+		if row.MinEpochStart < epochMin {
+			epochMin = row.MinEpochStart
+		}
+		if row.MaxEpochEnd > epochMax {
+			epochMax = row.MaxEpochEnd
+		}
+	}
 	// ------------------------------------------------------------------------------------------------------------------
 	// Get the EL rewards
```

Comment:
`validator_dashboard_view_execution_rewards_epoch` is a proxied view that exists in clickhouse that exposes `value` `proposer` and `epoch` - in theory this can even be combined in the above query.


but not really important enough to have this block the pr

Reply:
I will keep it in mind for the future.